akka-stream/src/main/scala/akka/stream/javadsl/Hub.scala

```diff
@@ -0,0 +1,71 @@
+package akka.stream.javadsl
+
+import akka.NotUsed
+
+/**
+ * A MergeHub is a special streaming hub that is able to collect streamed elements from a dynamic set of
+ * producers. It consists of two parts, a [[Source]] and a [[Sink]]. The [[Source]] streams the element to a consumer from
+ * its merged inputs. Once the consumer has been materialized, the [[Source]] returns a materialized value which is
+ * the corresponding [[Sink]]. This [[Sink]] can then be materialized arbitrary many times, where each of the new
+ * materializations will feed its consumed elements to the original [[Source]].
+ */
+object MergeHub {
+
+  /**
+   * Creates a [[Source]] that emits elements merged from a dynamic set of producers. After the [[Source]] returned
+   * by this method is materialized, it returns a [[Sink]] as a materialized value. This [[Sink]] can be materialized
+   * arbitrary many times and each of the materializations will feed the elements into the original [[Source]].
+   *
+   * Every new materialization of the [[Source]] results in a new, independent hub, which materializes to its own
+   * [[Sink]] for feeding that materialization.
+   *
+   * If one of the inputs fails the [[Sink]], the [[Source]] is failed in turn (possibly jumping over already buffered
+   * elements). Completed [[Sink]]s are simply removed. Once the [[Source]] is cancelled, the Hub is considered closed
+   * and any new producers using the [[Sink]] will be cancelled.
+   *
+   * @param clazz Type of elements this hub emits and consumes
+   * @param perProducerBufferSize Buffer space used per producer.
+   */
+  def of[T](clazz: Class[T], perProducerBufferSize: Int): Source[T, Sink[T, NotUsed]] = {
```

Comment:
Why does the java side need the Class, if it's never used, isn't the type parameter sufficient?
Also, why the difference in names vs the Scala API for the factory methods?